src/Actions/RemoveStorageSymlinksAction.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Actions;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\LazyCollection;
+use Stancl\Tenancy\Concerns\DealsWithTenantSymlinks;
+use Stancl\Tenancy\Database\Models\Tenant;
+use Stancl\Tenancy\Events\RemovingStorageSymlink;
+use Stancl\Tenancy\Events\StorageSymlinkRemoved;
+
+class RemoveStorageSymlinksAction
+{
+    use DealsWithTenantSymlinks;
+
+    public static function handle(Tenant|Collection|LazyCollection $tenants): void
+    {
+        $tenants = $tenants instanceof Tenant ? collect([$tenants]) : $tenants;
+
+        /** @var Tenant $tenant */
+        foreach ($tenants as $tenant) {
+            foreach (static::possibleTenantSymlinks($tenant) as $publicPath => $storagePath) {
+                static::removeLink($publicPath, $tenant);
+            }
+        }
+    }
+
+    protected static function removeLink(string $publicPath, Tenant $tenant): void
+    {
+        if (static::symlinkExists($publicPath)) {
+            event(new RemovingStorageSymlink($tenant));
+
+            app()->make('files')->delete($publicPath); // todo change this to use DI
```

Comment:
Do we plan on changing the class to be non-static?

Reply:
I think we don't. So even if `app()->make('files')` isn't ideal, we probably won't ever change it? Therefore the `todo` could be removed.

Reply:
I unresolved this, and am now trying to use DI here. I tried adding `protected Filesystem $filesystem` to the constructor so the action could have a `$filesystem` property, but I either get an error because the JobPipeline passed the action a `Tenant` instance – that's when I make `Filesystem` the first parameter. When I make it the second parameter, I get an error saying there are not enough parameters (`Filesystem` doesn't get resolved, but I'm not getting the 'unresolvable dependency' error that I was getting in the past). 

Making `handle()` accept `Filesystem` works, but only when the action is used in a JobPipeline, e.g. if we'd want to use the action, we'd have to do `(new RemoveStorageSymlinksAction($tenants))->handle(app()->make('files'))`. @stancl, could you please hint at a better way to go about this? I mean ideally, when we don't pass anything to `handle()`, the `Filesystem` would get resolved automatically, right?

Reply:
The easiest solution would be to revert the changes and make it static again. I simply *asked* if the point of the job classes is to be able to use the action in the pipeline given that the action is static, and if we could try to make the action work. If we can't, it's best to create a separate class and keep the code clean.

Reply:
Okay, I see. I'll revert the changes.